pkg/controller/bundle_sync_task.go

```diff
@@ -212,33 +212,48 @@ func (st *bundleSyncTask) handleProcessResult(retriable bool, processErr error)
-		case resourceStatusBlockedByError:
-			blockedCond.Status = smith_v1.ConditionTrue
-			blockedCond.Reason = smith_v1.ResourceReasonOtherResourceError
-			blockedCond.Message = "Some other resource is in error state"
-		case resourceStatusInProgress:
-			inProgressCond.Status = smith_v1.ConditionTrue
-		case resourceStatusReady:
-			readyCond.Status = smith_v1.ConditionTrue
-		case resourceStatusError:
-			errorCond.Status = smith_v1.ConditionTrue
-			errorCond.Message = resStatus.err.Error()
-			if resStatus.isRetriableError {
-				errorCond.Reason = smith_v1.ResourceReasonRetriableError
+		if resInfo, ok := st.processedResources[res.Name]; ok {
+			// Resource was processed
+			switch resStatus := resInfo.status.(type) {
+			case resourceStatusDependenciesNotReady:
+				blockedCond.Status = smith_v1.ConditionTrue
+				blockedCond.Reason = smith_v1.ResourceReasonDependenciesNotReady
+				blockedCond.Message = fmt.Sprintf("Not ready: %q", resStatus.dependencies)
+			case resourceStatusBlockedByError:
+				blockedCond.Status = smith_v1.ConditionTrue
+				blockedCond.Reason = smith_v1.ResourceReasonOtherResourceError
+				blockedCond.Message = "Some other resource is in error state"
+			case resourceStatusInProgress:
 				inProgressCond.Status = smith_v1.ConditionTrue
-			} else {
+			case resourceStatusReady:
+				readyCond.Status = smith_v1.ConditionTrue
+			case resourceStatusError:
+				errorCond.Status = smith_v1.ConditionTrue
+				errorCond.Message = resStatus.err.Error()
+				if resStatus.isRetriableError {
+					errorCond.Reason = smith_v1.ResourceReasonRetriableError
+					inProgressCond.Status = smith_v1.ConditionTrue
+				} else {
+					errorCond.Reason = smith_v1.ResourceReasonTerminalError
+				}
+				failedResources = append(failedResources, res.Name)
+				retriableResourceErr = retriableResourceErr && resStatus.isRetriableError // Must not continue if at least one error is not retriable
+			default:
+				blockedCond.Status = smith_v1.ConditionUnknown
+				inProgressCond.Status = smith_v1.ConditionUnknown
+				readyCond.Status = smith_v1.ConditionUnknown
+				errorCond.Status = smith_v1.ConditionTrue
 				errorCond.Reason = smith_v1.ResourceReasonTerminalError
+				errorCond.Message = fmt.Sprintf("internal error - unknown resource status type %T", resInfo.status)
+				failedResources = append(failedResources, res.Name)
+				retriableResourceErr = false
 			}
-			failedResources = append(failedResources, res.Name)
-			retriableResourceErr = retriableResourceErr && resStatus.isRetriableError // Must not continue if at least one error is not retriable
-		default:
-			panic(errors.Errorf("unknown resource status type %T", resInfo.status))
+		} else {
+			// Resource was not processed
+			blockedCond.Status = smith_v1.ConditionUnknown
+			inProgressCond.Status = smith_v1.ConditionUnknown
+			readyCond.Status = smith_v1.ConditionUnknown
+			errorCond.Status = smith_v1.ConditionUnknown
```

Comment:
Do we have enough error reporting in other places in case resource was not processed?
Thinking whether we need to set `errorCond.Status = smith_v1.ConditionTrue` and `errorCond.Message = ...` here.

Reply:
Yes. The only reason a resource might have not been processed if there was an error. That error is passed via `processErr` and is reported via bundle's condition.